be/src/storage/lake/vacuum.cpp

```diff
@@ -393,7 +394,7 @@ static Status vacuum_tablet_metadata(TabletManager* tablet_mgr, std::string_view
                                                 &tablet_vacuumed_version));
         RETURN_IF_ERROR(datafile_deleter.finish());
         RETURN_IF_ERROR(metafile_deleter.finish());
-        if (*vacuumed_version == 0 || *vacuumed_version > tablet_vacuumed_version) {
+        if (*vacuumed_version == -1 || *vacuumed_version > tablet_vacuumed_version) {
```

Comment:
this `-1` comes from the upper level initial value, which is easily overlooked. Shall not depends on the initial value set outside. Use its own logic to determine whether need to update the `vacuumed_version` to `tablet_vacuumed_version`.

Inside this function, it is only safe to assume that the `tablet_vacuumed_version` is set to `0` if not modified by `collect_files_to_vacuum()`

Reply:
> this `-1` comes from the upper level initial value, which is easily overlooked. Shall not depends on the initial value set outside. Use its own logic to determine whether need to update the `vacuumed_version` to `tablet_vacuumed_version`.
> 
> Inside this function, it is only safe to assume that the `tablet_vacuumed_version` is set to `0` if not modified by `collect_files_to_vacuum()`

Good suggestion, depends on the upper level initial value is bad.